In [1]:
import pandas as pd
import altair as alt
alt.renderers.enable('mimetype')

RendererRegistry.enable('mimetype')

In [2]:
resenas = pd.read_csv('../results/all_amazon_reviews.csv')

In [3]:
resenas['star_rating'].value_counts()

star_rating
5.0 de 5 estrellas    203
No rating              95
4.0 de 5 estrellas     47
1.0 de 5 estrellas     22
2.0 de 5 estrellas      7
3.0 de 5 estrellas      6
Name: count, dtype: int64

In [4]:
base = alt.Chart(resenas).mark_bar().encode(alt.X("star_rating", title ='Estrellas'), 
    alt.Y('count()', title = 'Conteo'), alt.Color("star_rating:N"))

# barras de la gráfica
bar = base.mark_bar().properties(width=500,height=300,
    title='Conteo de calificacion por cantidad de estrellas')
# texto de la gráfica
text = base.mark_text(baseline='bottom').encode(
    alt.Text('count()', format=",.0f"))

# poner barras y texto de la gráfica juntos
bar + text

<VegaLite 5 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/display_frontends.html#troubleshooting


In [5]:
#Creamos un DF con los Value_count y agregamos una columna con el porcentaje que representa cada conteo
resenas_count = resenas['star_rating'].value_counts().reset_index()
resenas_count.columns = ['star_rating','Count']
resenas_count['Porcentaje'] = (resenas_count['Count'] / resenas_count['Count'].sum()) * 100
resenas_count

star_rating  Count  Porcentaje
0  5.0 de 5 estrellas    203   53.421053
1           No rating     95   25.000000
2  4.0 de 5 estrellas     47   12.368421
3  1.0 de 5 estrellas     22    5.789474
4  2.0 de 5 estrellas      7    1.842105
5  3.0 de 5 estrellas      6    1.578947

#### De la gráfica y de este análisis podemos ver que existen 53 % de reseñas que dan excelente calificación con el máximo de 5 estrellas.
#### Un 12% da muy buena calificación con 4 estrellas. 
#### Un 5.7 % da una calificación Media con 3 estrellas
#### Un 7.5% da calificación Mala (1.8%)  y Muy  mala (5.7%) con 2 y 1 estrellas respectivamente.
#### Y un 25% no puso calificación con estrellas en su reseña lo cual no es posible saber en base a estrellas si su reseña es buena o mala y puede caer en cualquier categoría. 

# Vamos a limpiar (eliminar) las filas que no tinen rating y ver que tanta diferencia hay

In [6]:
resenas = resenas[resenas['star_rating'] != 'No rating']
resenas['star_rating'].value_counts()

star_rating
5.0 de 5 estrellas    203
4.0 de 5 estrellas     47
1.0 de 5 estrellas     22
2.0 de 5 estrellas      7
3.0 de 5 estrellas      6
Name: count, dtype: int64

In [7]:
base = alt.Chart(resenas).mark_bar().encode(alt.X("star_rating", title ='Estrellas'), 
    alt.Y('count()', title = 'Conteo'), alt.Color("star_rating:N"))

# barras de la gráfica
bar = base.mark_bar().properties(width=500,height=300,
    title='Conteo de calificacion por cantidad de estrellas')
# texto de la gráfica
text = base.mark_text(baseline='bottom').encode(
    alt.Text('count()', format=",.0f"))

# poner barras y texto de la gráfica juntos
bar + text

<VegaLite 5 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/display_frontends.html#troubleshooting


In [8]:
#Creamos un DF con los Value_count y agregamos una columna con el porcentaje que representa cada conteo
resenas_count = resenas['star_rating'].value_counts().reset_index()
resenas_count.columns = ['star_rating','Count']
resenas_count['Porcentaje'] = (resenas_count['Count'] / resenas_count['Count'].sum()) * 100
resenas_count

star_rating  Count  Porcentaje
0  5.0 de 5 estrellas    203   71.228070
1  4.0 de 5 estrellas     47   16.491228
2  1.0 de 5 estrellas     22    7.719298
3  2.0 de 5 estrellas      7    2.456140
4  3.0 de 5 estrellas      6    2.105263

#### Al elmiminar las filas sin rating obtenemos un 71 % de reseñas que dan excelente calificación con el máximo de 5 estrellas.
#### Un 16.5% da muy buena calificación con 4 estrellas. 
#### Un 2 % da una calificación Media con 3 estrellas
#### Un 10% da calificación Mala (2.4%)  y Muy  mala (7.7%) con 2 y 1 estrellas respectivamente.